markdown.go

```diff
@@ -58,6 +61,16 @@ func cleanupMarkDown(mdString string) (md string, anchors []string) {
 	mdString = strings.ReplaceAll(mdString, "\t", "    ")
 	mdString = strings.ReplaceAll(mdString, "https://docs.docker.com", "")
 
+	// Rewrite internal links, replacing relative paths with absolute path
+	// e.g. from [docker buildx build](buildx_build.md#build-arg)
+	// to [docker buildx build](/reference/cli/docker/buildx/build/#build-arg)
+	mdString = relativeLink.ReplaceAllStringFunc(mdString, func(link string) string {
```

Comment:
It's still a bit unclear to my why (assuming we can move `buildx_build.md` to `buildx/build.md`) linking to that wouldn't work 🤔 You mentioned having to add `_index.md` files, but wouldn't those be created in our docs repository (i.e., docs repository uses the YAML files, correct?)

Reply:
it doesn't work because of the intermediate commands. E.g. `docker manifest` is `docker/manifest/_index.md` in the docs repo. Unless we also want use that directory and filename structure in the upstream repositories. But that would require changes to how we generate markdown files (https://github.com/docker/cli-docs-tool/blob/main/clidocstool_md.go#L138-L144) and how we get the markdown source for yaml generation (https://github.com/docker/cli-docs-tool/blob/main/clidocstool_yaml.go#L390).

Or we could have intermediate commands live in the directory of the parent command, e.g. `docs/reference/commandline/manifest.md`, and its subcommands in e.g. `docs/reference/commandline/manifest/create.md`. But this structure looks a bit weird to me.

I guess it's probably subjective but having the reference docs in a flat structure seems like an easy way to manage them. We're not pulling those files as-is anyway, and we have this preprocessing step, so it seemed to me like a reasonable place to handle it.